BEST_PRACTICE.md

```diff
@@ -0,0 +1,247 @@
+# Clib Examples and Best Practice
+
+This page will cover:
+
+ - [How to use libraries](#how-to-use-installed-libraries).
+ - [Example Makefile](#example-makefile).
+ - [Example `package.json` for executables](#example-packagejson-for-exacutables).
+ - [Making your own library package](#making-your-own-libraries).
+ - [Example `package.json` for libraries](#example-packagejson-for-libraries).
+ - [How to install/uninstall executables](#install-and-uninstall-executables).
+
+For instructions on installation, click [here](https://github.com/clibs/clib#installation).
+
+## How to use installed libraries for your project
+
+Lets say you have your project, with a typical directory tree like this:
+
+```
+your-project/
+│
+├── deps/
+│   ├── trim.c/
+│   │   ├── trim.h
+│   │   ├── trim.c
+│   │   └── package.json
+│   │
+│   ├── commander/
+│   │   ├─ commander.h
+│   │   ├─ commander.c
+│   │   └─ package.json
+│   │
+│   └── logger/
+│       ├── logger.h
+│       ├── logger.c
+│       └── package.json
+│
+├── LICENSE
+│
+├── Makefile
+│
+├── README.md
+│
+├── package.json
+│
+├── src/
+│   ├── main.c
+│   ├── function.c
+│   ├── function.h
+│   └── etc...
+│
+└── test.sh
+```
+
+`test.sh` is your test script. `src` is the directory where you source
```

Comment:
Not many projects use a `test.sh` script. Tests are more commonly written in C.

Reply:
Okay, I'll remove it :+1: